cluster/src/dunit/scala/io/snappydata/cluster/QueryRoutingDUnitSecureTest.scala

```diff
@@ -0,0 +1,402 @@
+      })
+
+      // scalastyle:off println
+      println(s"insertRows2: committed $numRows rows")
+      // scalastyle:on println
+    } finally {
+      stmt1.close()
+      conn.close()
+    }
+  }
+
+  def insertRows3(numRows: Int, serverHostPort: Int, tableName: String,
+      user: String, pass: String): Unit = {
+    val conn = netConnection(serverHostPort, user, pass)
+    // scalastyle:off println
+    println(s"insertRows3: Connected to $serverHostPort")
+    // scalastyle:on println
+
+    val stmt1 = conn.createStatement()
+    try {
+      var i = 1
+      (1 to numRows).foreach(_ => {
+        stmt1.addBatch(s"insert into $tableName values($i, $i, '$i')")
+        i += 1
+        if (i % 10 == 0) {
+          stmt1.executeBatch()
+          i = 0
+        }
+      })
+      stmt1.executeBatch()
+
+      // scalastyle:off println
+      println(s"insertRows3: committed $numRows rows")
+      // scalastyle:on println
+    } finally {
+      stmt1.close()
+      conn.close()
+    }
+  }
+
+  def insertRows4(numRows: Int, serverHostPort: Int, tableName: String,
+      user: String, pass: String): Unit = {
+    val conn = netConnection(serverHostPort, user, pass)
+    // scalastyle:off println
+    println(s"insertRows4: Connected to $serverHostPort")
+    // scalastyle:on println
+
+    val stmt1 = conn.createStatement()
+    try {
+      (1 to numRows).foreach(i => {
+        stmt1.executeUpdate(s"insert into $tableName values($i, $i, '$i')")
+      })
+
+      // scalastyle:off println
+      println(s"insertRows4: committed $numRows rows")
+      // scalastyle:on println
+    } finally {
+      stmt1.close()
+      conn.close()
+    }
+  }
+
+  def verifyQuery1(qryTest: String, stmt_rs: ResultSet): Unit = {
+    val builder = StringBuilder.newBuilder
+
+    var index = 0
+    while (stmt_rs.next()) {
+      index += 1
+      val stmt_i = stmt_rs.getInt(1)
+      val stmt_j = stmt_rs.getInt(2)
+      val stmt_s = stmt_rs.getString(3)
+      if (index % 100 == 0) {
+        builder.append(s"$qryTest Stmt: row($index) $stmt_i $stmt_j $stmt_s ").append("\n")
+      }
+    }
+    builder.append(s"$qryTest Stmt: Total number of rows = $index").append("\n")
+    // scalastyle:off println
+    println(builder.toString())
+    // scalastyle:on println
+    assert(index == 4000)
+  }
+
+  def verifyQuery2(qryTest: String, stmt_rs: ResultSet): Unit = {
+    val builder = StringBuilder.newBuilder
+
+    var index = 0
+    while (stmt_rs.next()) {
+      index += 1
+      val stmt_i = stmt_rs.getInt(1)
+      val stmt_j = stmt_rs.getInt(2)
+      val stmt_s = stmt_rs.getString(3)
+      if (index % 10 == 0) {
+        builder.append(s"$qryTest Stmt: row($index) $stmt_i $stmt_j $stmt_s ").append("\n")
+      }
+    }
+    builder.append(s"$qryTest Stmt: Total number of rows = $index").append("\n")
+    // scalastyle:off println
+    println(builder.toString())
+    // scalastyle:on println
+    assert(index == 400)
+  }
+
```

Comment:
verifyQuery1 and verifyQuery2 look same except for num of rowss verified(var index)  which can be paramterized and we can have one function only. Probably we can have just one function.